src/main/java/com/redhat/devtools/lsp4ij/features/selectionRange/LSPCodeBlockExtendWordSelectionHandler.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.selectionRange;
+
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.containers.ContainerUtil;
+import com.intellij.util.text.TextRangeUtil;
+import com.redhat.devtools.lsp4ij.features.codeBlockProvider.LSPCodeBlockProvider;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Composite extend word selection handler for contents of LSP-defined code blocks that ensures that we only have to
+ * retrieve the code block text range once for all constituent selectioners.
+ */
+public class LSPCodeBlockExtendWordSelectionHandler extends AbstractLSPExtendWordSelectionHandler {
```

Comment:
This is the one `extendWordSelectionHandler` implementation for code block contents. It's implemented as a composite to ensure that we only have to call `LSPCodeBlockProvider.getCodeBlockRange()` once for all constituent selectioners instead of once per-selectioner.